tests/lib/rules/use-waiters-in-module-scope.js

```diff
@@ -0,0 +1,55 @@
+'use strict';
+
+const rule = require('../../../lib/rules/use-waiters-in-module-scope');
+
+const { ERROR_MESSAGE } = rule;
+const RuleTester = require('eslint').RuleTester;
+
+const ruleTester = new RuleTester({
+  parserOptions: {
+    ecmaVersion: 6,
+    sourceType: 'module',
+  },
+});
+
+ruleTester.run('use-waiters-in-module-scope', rule, {
+  valid: [
+    `
+    import { buildWaiter } from 'ember-test-waiters';
+
+    let myWaiter = buildWaiter('waiterName');
+  `,
+    `
+    import { buildWaiter } from 'table-waiters';
+
+    function useWaiter() {
+      let myOtherWaiter = buildWaiter('the second');
+    }
+  `,
```

Comment:
This should be a valid case:

```
import { buildWaiter } from 'ember-test-waiters';

const myWaiter = someThing(otherThing(buildWaiter()));
```

Reply:
Again, this would be against best practices to provide this type of abstraction over top of the waiter system. It's absolutely not intended to be used this way, and this would be considered an abuse of the library. We may want to extend the lint rule to catch this. cc @rwjblue

Reply:
See my reply in other thread: https://github.com/ember-cli/eslint-plugin-ember/pull/702#discussion_r391107587

Reply:
Resolving as per my reply in your thread.